src/components/SearchAndFilter/SearchAndFilter.tsx

```diff
@@ -261,7 +261,11 @@ const SearchAndFilter = ({
               lead={chip.lead}
               value={chip.value}
               key={`search-${chip.lead}+${chip.value}`}
-              onDismiss={() => removeFromSelected(chip)}
+              onDismiss={(event) => {
+                // Prevent filter chip dismissals from bubbling up and triggering the parent onClick handler
+                event.stopPropagation();
```

Comment:
It might be a good idea to add a test for this change to catch this issue if someone updates this code.

Reply:
Test has been added

Reply:
Thanks!